src/symmetry.c

```diff
@@ -959,14 +959,37 @@ static PointSymmetry get_lattice_symmetry(const Cell *cell,
         for (i = 0; i < 26; i++) {
             for (j = 0; j < 26; j++) {
                 for (k = 0; k < 26; k++) {
-                    /* For layer group, some rotations are not permitted. */
-                    if (aperiodic_axis != -1 &&
-                        ((aperiodic_axis == 2 && k != 2 && k != 5) ||
-                         (aperiodic_axis == 0 && i != 0 && i != 3) ||
-                         (aperiodic_axis == 1 && j != 1 && j != 4))) {
-                        continue;
-                    }
                     set_axes(axes, i, j, k);
+                    /* For layer groups, the off-diagonal elements for the
+                     * aperiodic axis are zero. */
+                    switch (aperiodic_axis) {
+                        case 2:
+                            /*    {{W_11, W_12,      0}, *
+                             * W = {W_21, W_22,      0}, *
+                             *     {   0,    0, (+/-)1}} */
+                            if (axes[0][2] || axes[1][2] || axes[2][0] ||
+                                axes[2][1])
+                                continue;
+                            break;
+                        case 0:
```

Comment:
Readability wise much better, I will defer to @lan496 to check if these are equivalent.

Small note though, all switch cases are equivalent, so you can compress them bi avoiding the `continue:break;` in the middle.

Reply:
I am a little confused with how to do it? As the if-clause in each switch case are different?

Reply:
Ah, sorry, forgot about th `if` statements in between. Yeah, there  is no clean way of doing so. At least not within C, with C++ it could have been simplified with templated function.

Reply:
I see. Thank you.